controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -239,14 +242,32 @@ func (r *StorageClassRequestReconciler) initPhase(storageProfile *v1.StorageProf
 	} else if r.StorageClassRequest.Spec.Type == "sharedfilesystem" {
 		r.cephFilesystemSubVolumeGroup = &rookCephv1.CephFilesystemSubVolumeGroup{}
 		r.cephFilesystemSubVolumeGroup.Namespace = r.OperatorNamespace
-		for _, res := range r.StorageClassRequest.Status.CephResources {
-			if res.Kind == "CephFilesystemSubVolumeGroup" {
-				r.cephFilesystemSubVolumeGroup.Name = res.Name
-				break
-			}
+
+		cephFilesystemSubVolumeGroupList := &rookCephv1.CephFilesystemSubVolumeGroupList{}
+		err := r.Client.List(r.ctx, cephFilesystemSubVolumeGroupList, client.InNamespace(r.OperatorNamespace))
+		if err != nil {
+			return err
 		}
-		if r.cephFilesystemSubVolumeGroup.Name == "" {
-			r.cephFilesystemSubVolumeGroup.Name = fmt.Sprintf("cephfilesystemsubvolumegroup-%s-%s", r.storageConsumer.Name, generateUUID())
+		ownedCephFilesystemSubVolumeGroups := util.Filter(cephFilesystemSubVolumeGroupList.Items, func(item *rookCephv1.CephFilesystemSubVolumeGroup) bool {
+			for _, ownerReference := range item.OwnerReferences {
+				if ownerReference.UID == r.StorageClassRequest.UID {
+					return true
+				}
+			}
+			return false
+		})
```

Comment:
1. Use multiline formatting when the line is long
2. Use pointers when iterating on complex data types to prevent deep copy of the object. Or if you only need to access the item once, refer to it by index 

```suggestion
		ownedCephFilesystemSubVolumeGroups := util.Filter(
			cephFilesystemSubVolumeGroupList.Items, 
			func(item *rookCephv1.CephFilesystemSubVolumeGroup) bool {
				for i := range item.OwnerReferences {
					if item.OwnerReferences[i].UID == r.StorageClassRequest.UID {
						return true
					}
				}
				return false
			}
		)
```